src/catalog/database_catalog.cpp

```diff
@@ -637,6 +670,21 @@ table_oid_t DatabaseCatalog::CreateTable(transaction::TransactionContext *const
   return CreateTableEntry(txn, table_oid, ns, name, schema) ? table_oid : INVALID_TABLE_OID;
 }
 
+bool DatabaseCatalog::DeleteIndexes(transaction::TransactionContext *const txn, const table_oid_t table) {
+  if (!TryLock(txn)) return false;
+  // Get the indexes
+  const auto index_oids = GetIndexOids(txn, table);
+  // Delete all indexes
+  for (const auto index_oid : index_oids) {
+    auto result = DeleteIndex(txn, index_oid);
+    if (!result) {
+      // write-write conflict. Someone beat us to this operation.
```

Comment:
Is this possible with the database-level DDL lock?

Reply:
It is not. But since the DDL lock is a possible temporary hack, I figured I'd be as consistent with existing logic as possible.